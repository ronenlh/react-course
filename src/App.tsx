import React, { useEffect, useState } from 'react';
import { Route, Routes } from 'react-router-dom';
import logo from './logo.png';
import './App.css';
import ProductList from './components/ProductList';
import Navigation from './components/Navigation';
import ProductDetails from './components/ProductDetails';
import Lost from './components/Lost';
import data from './data/data.json';
import Loading from './components/Loading';

const Contact = React.lazy(() => import('./components/Contact'));

const App = () => {
  const [isToggled, setToggled] = useState(true);
  const [isNavOpen, setNavOpen] = useState(false);
  const [cards, setCards] = useState(data);

  useEffect(() => {
      setCards(data);
  }, []);

  useEffect(() => {
    const nav = document.getElementById('myNav');
    if (!nav) return;

    if (isNavOpen) {
      nav.style.width = "100%";
    } else {
      nav.style.width = "0%";
    }
  }, [isNavOpen]);

  return (
    <div className="App">
      <header className="App-header">
        <img src={logo}
             className={isToggled ? 'static-logo' : 'static-logo animated jello'}
             alt="logo"
             onMouseEnter={() => setToggled(!isToggled)}
             onMouseLeave={() => setToggled(!isToggled)}
             onClick={() => setNavOpen(true)} />
        <h1 className={isToggled ? 'menu-hidden' : 'menu animated bounceInDown'}
            onClick={() => setNavOpen(true)}
        >Menu</h1>
        <Navigation closeNav={() => setNavOpen(false)} />
      </header>

      <Routes>
        <Route path="/" element={<ProductList cards={cards} />} />
        <Route path="/supplements" element={
          <ProductList cards={cards.filter(({ type }) => type === 'supplement')} />}
        />
        <Route path="/clothing" element={
          <ProductList cards={cards.filter(({ type }) => type === 'clothing')} />}
        />
        <Route path="/vitamin" element={
          <ProductList cards={cards.filter(({ type }) => type === 'vitamin')} />}
        />
        <Route path="/contact" element={
          <React.Suspense fallback={<Loading />}>
            <Contact />
          </React.Suspense>} />
        <Route path="/product/:productId" element={<ProductDetails />} />
        <Route path="*" element={<Lost />} />
      </Routes>
    </div>
  );
};

export default App;
