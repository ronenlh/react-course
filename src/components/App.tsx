import Counter from './Counter';
import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { CounterState } from '../reducers';

const App: React.FC = () => {
    const count = useSelector<CounterState, number>((state) => state.counter);
    const dispatch = useDispatch()

    return (<Counter
        value={count}
        onIncrement={() => dispatch({ type: 'INCREMENT' })}
        onDecrement={() => dispatch({ type: 'DECREMENT' })}
    />);
};

export default App;
