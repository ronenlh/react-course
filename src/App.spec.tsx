import React from 'react';
import renderer from 'react-test-renderer';

import App from './App';
import AuthContextProvider from './context/auth-context';

it('renders correctly', () => {
    const app = renderer.create(<App />);

    const tree = app.toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly with AuthContextProvider', () => {
    const app = renderer.create(<AuthContextProvider>
        <App />
    </AuthContextProvider>);

    const tree = app.toJSON();
    expect(tree).toMatchSnapshot();
});
