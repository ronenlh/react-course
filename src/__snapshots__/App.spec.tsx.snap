// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="auth"
>
  <div
    className="card"
  >
    <h2>
      You are not authenticated!
    </h2>
    <p>
      Please log in to continue.
    </p>
    <button
      onClick={[Function]}
    >
      Log In
    </button>
  </div>
</div>
`;

exports[`renders correctly with AuthContextProvider 1`] = `
<div
  className="auth"
>
  <div
    className="card"
  >
    <h2>
      You are not authenticated!
    </h2>
    <p>
      Please log in to continue.
    </p>
    <button
      onClick={[Function]}
    >
      Log In
    </button>
  </div>
</div>
`;
